tests/Greenspun/REPL.cpp

```diff
@@ -0,0 +1,254 @@
+
+#include <iostream>
+
+
+#include "Pregel/Algos/AIR/Greenspun/Interpreter.h"
+#include "Pregel/Algos/AIR/Greenspun/Primitives.h"
+#include "Pregel/Algos/AIR/Greenspun/EvalResult.h"
+#include "velocypack/Builder.h"
+#include "velocypack/Parser.h"
+#include "velocypack/velocypack-aliases.h"
+
+#include "Utilities/Completer.h"
+#include "Utilities/LineEditor.h"
+#include "Utilities/ShellBase.h"
+
+/* YOLO */
+
+using namespace arangodb;
+
+greenspun::EvalResult Func_thisId(greenspun::Machine& ctx, VPackSlice const params, VPackBuilder& result) {
+  result.add(VPackValue("V/1"));
+  return {};
+}
+
+std::unordered_map<std::string, VPackBuilder> variableValues;
+
+greenspun::EvalResult Func_VarSet(greenspun::Machine& ctx, VPackSlice const params, VPackBuilder& result) {
+  if (params.length() != 2) {
+    return greenspun::EvalError("expected two parameters");
+  }
+  VPackSlice name = params.at(0);
+  VPackSlice value = params.at(1);
+  if (!name.isString()) {
+    return greenspun::EvalError("expected string as first parameter, found: " + name.toJson());
+  }
+
+  auto nameStr = name.copyString();
+  VPackBuilder& builder = variableValues[nameStr];
+  builder.clear();
+  builder.add(value);
+
+  ctx.setVariable(nameStr, builder.slice());
+  return {};
+}
+
+void AddSomeFunctions(greenspun::Machine& m) {
+  m.setFunction("this-id", Func_thisId);
+  m.setFunction("var-set", Func_VarSet);
+}
+
+struct LispCompleter : arangodb::Completer {
+  bool isComplete(std::string const& source, size_t /*lineno*/) final {
```

Comment:
@goedderz is right, extracting `_id` from an ArangoDB document requires special handling, because the value of `_id` is not a string but of `ValueType::Custom`. This needs to be either std::stringified or passed on as a custom value, and then whoever interprets it must handle it.

Reply:
Removed all validation cases related to JS - so from now on, only JSON is supported.